### PR TITLE
feat(pub_sub/go/sdk): Upgrade Dapr Go SDK to 1.4.0

### PR DESCRIPTION
**Pull request by noahziheng**
# Description

- Upgrade Dapr Go SDK to 1.4.0 for pub_sub Quick Start
- SDK example changed for 1.4.0
  - Subscription/Route need change to `/orders` for avoid 404 exception
  - remove useless `http.HandleFunc` implemention 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] The quickstart code compiles correctly
* [x] You've tested new builds of the quickstart if you changed quickstart code
* [x] You've updated the quickstart's README if necessary
* [ ] If you have changed the steps for a quickstart be sure that you have updated the automated validation accordingly. All of our quickstarts have annotations that allow them to be executed automatically as code. For more information see [mechanical-markdown](https://github.com/dapr/mechanical-markdown#readme). For user guide with examples see [Examples](https://github.com/dapr/mechanical-markdown/tree/main/examples#mechanical-markdown-by-example).

### CODE REVIEW

File: pub_sub/go/sdk/checkout/go.mod [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
 
-go 1.17
+go 1.18
+
+require github.com/dapr/go-sdk v1.4.0
 
 require (
-	github.com/dapr/go-sdk v1.2.0 // indirect
-	github.com/golang/protobuf v1.4.3 // indirect
+	github.com/dapr/dapr v1.6.1-0.20220223052122-c503710ab2d1 // indirect
```
Comment (msfussell) [inline pub_sub/go/sdk/checkout/go.mod:8]: Do you need to pull in dapr v1.6.1 into this as a dependency? And if so, best move to v1.7.2

File: pub_sub/go/sdk/checkout/go.sum [hunk 2]
```diff
@@ -1,96 +1,138 @@
...
+github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/client9/misspell v0.3.4/go.mod h1:qj6jICC3Q7zFZvVWo7KLAzC3yx5G7kyvSDkc90ppPyw=
-github.com/cncf/udpa/go v0.0.0-20200629203442-efcf912fb354/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
-github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
-github.com/dapr/go-sdk v1.2.0 h1:9Uvw3AJlgfEPBDpaFg5zThe9dqv+ag+9AzFAiDZNYHo=
-github.com/dapr/go-sdk v1.2.0/go.mod h1:zyhsocIKv4pqQ2VtvWvf2CK1UhP7Z2OAOXgEpVxMgIs=
+github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f/go.mod h1:M8M6+tZqaGXZJjfX53e64911xZQV5JYwmTeXPW+k8Sc=
+github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
+github.com/cncf/xds/go v0.0.0-20210312221358-fbca930ec8ed/go.mod h1:eXthEFrGJvWHgFFCl3hGmgk+/aYT6PnTQLykKQRLhEs=
+github.com/dapr/dapr v1.6.1-0.20220223052122-c503710ab2d1 h1:Yuvx+yFFrYeQym9JptSO7ur/qi1J9veuNI267LwE7uk=
```
Comment (msfussell) [inline pub_sub/go/sdk/checkout/go.sum:12]: Do you need to pull in dapr v1.6.1 into this as a dependency? And if so, best move to v1.7.2

File: pub_sub/go/sdk/checkout/go.sum [hunk 3]
```diff
@@ -1,96 +1,138 @@
...
-github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
-github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e/go.mod h1:zD1mROLANZcx1PVRCS0qkT7pwLkGfwJo4zjcN/Tysno=
+github.com/google/uuid v1.3.0 h1:t6JiXgmwXMjEs8VusXIJk2BXHsn+wx8BZdTaoZ5fu7I=
+github.com/grpc-ecosystem/grpc-gateway v1.16.0/go.mod h1:BDjrQk3hbvj6Nolgz8mAMFbcEtjT1g+wF4CSlocrBnw=
 github.com/pkg/errors v0.9.1 h1:FEBLx1zS214owpjy7qsBeixbURkuhQAwrK5UwLGTwt4=
 github.com/pkg/errors v0.9.1/go.mod h1:bwawxfHBFNV+L2hUp1rHADufV3IMtnDRdf1r5NINEl0=
+github.com/pmezard/go-difflib v1.0.0 h1:4DBwDE0NGyQoBHbLQYPwSUPoCMWR5BEzIk/f1lZbAQM=
 github.com/pmezard/go-difflib v1.0.0/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4/go.mod h1:xMI15A0UPsDsEKsMN9yxemIoYk6Tm2C1GtYGdfGttqA=
+github.com/rogpeppe/fastuuid v1.2.0/go.mod h1:jVj6XXZzXRy/MSR5jhDC/2q6DgLz+nrA6LYCDYWNEvQ=
```
Comment (msfussell) [inline pub_sub/go/sdk/checkout/go.sum:56]: There are several new dependencies added here. Given the code has not changed, any reason these are being included now?

File: pub_sub/go/sdk/order-processor/go.mod [hunk 4]
```diff
@@ -1,16 +1,19 @@
...
 
-go 1.17
+go 1.18
+
+require github.com/dapr/go-sdk v1.4.0
 
 require (
-	github.com/dapr/go-sdk v1.2.0 // indirect
-	github.com/golang/protobuf v1.4.3 // indirect
+	github.com/dapr/dapr v1.6.1-0.20220223052122-c503710ab2d1 // indirect
```
Comment (msfussell) [inline pub_sub/go/sdk/order-processor/go.mod:8]: Do you need to pull in dapr v1.6.1 into this as a dependency? And if so, best move to v1.7.2

File: pub_sub/go/sdk/order-processor/go.sum [hunk 5]
```diff
@@ -1,96 +1,141 @@
...
+github.com/cespare/xxhash v1.1.0/go.mod h1:XrSqR1VqqWfGrhpAt58auRo0WTKS1nRRg3ghfAqPWnc=
 github.com/client9/misspell v0.3.4/go.mod h1:qj6jICC3Q7zFZvVWo7KLAzC3yx5G7kyvSDkc90ppPyw=
-github.com/cncf/udpa/go v0.0.0-20200629203442-efcf912fb354/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
-github.com/creack/pty v1.1.9/go.mod h1:oKZEueFk5CKHvIhNR5MUki03XCEU+Q6VDXinZuGJ33E=
-github.com/dapr/go-sdk v1.2.0 h1:9Uvw3AJlgfEPBDpaFg5zThe9dqv+ag+9AzFAiDZNYHo=
-github.com/dapr/go-sdk v1.2.0/go.mod h1:zyhsocIKv4pqQ2VtvWvf2CK1UhP7Z2OAOXgEpVxMgIs=
+github.com/cncf/udpa/go v0.0.0-20191209042840-269d4d468f6f/go.mod h1:M8M6+tZqaGXZJjfX53e64911xZQV5JYwmTeXPW+k8Sc=
+github.com/cncf/udpa/go v0.0.0-20201120205902-5459f2c99403/go.mod h1:WmhPx2Nbnhtbo57+VJT5O0JRkEi1Wbu0z5j0R8u5Hbk=
+github.com/cncf/xds/go v0.0.0-20210312221358-fbca930ec8ed/go.mod h1:eXthEFrGJvWHgFFCl3hGmgk+/aYT6PnTQLykKQRLhEs=
+github.com/dapr/dapr v1.6.1-0.20220223052122-c503710ab2d1 h1:Yuvx+yFFrYeQym9JptSO7ur/qi1J9veuNI267LwE7uk=
```
Comment (msfussell) [inline pub_sub/go/sdk/order-processor/go.sum:12]: Do you need to pull in dapr v1.6.1 into this as a dependency? And if so, best move to v1.7.2

**Comment by msfussell** [comment]
@noahziheng - Thanks for catching these issue. I have posted a few questions above?

File: pub_sub/go/sdk/checkout/go.sum (hunk 3 above)
Comment (noahziheng) [reply pub_sub/go/sdk/checkout/go.sum:58]: I found github.com/rogpeppe/fastuuid is a indirect dependencies, imported by google.golang.org/grpc@v1.40.0.

so deep dependency tree....

image

**Comment by noahziheng** [comment]
@msfussell I had updated github.com/dapr/dapr to v1.7.2. 

In fact, the latest github.com/dapr/go-sdk@1.4.0 depend 1.6.1 now. Is that need bump to 1.7.2 too?

**Comment by msfussell** [comment]
> @msfussell I have updated github.com/dapr/dapr to v1.7.2.
> 
> In fact, the latest github.com/dapr/go-sdk@1.4.0 depends on 1.6.1 now. Is that need to be bumped to 1.7.2 too?

Yes, I would have expected this to pull in 1.7.2. Can you raise and issue in the Go SDK for this?

**Review by msfussell — APPROVED** [review]
LGTM

**Comment by msfussell** [comment]
@noahziheng - Looks like there are failures on the validation. PTAL.

**Comment by noahziheng** [comment]
@msfussell I have fixed this issue of validating, the actions need re-run by approval.

**Comment by noahziheng** [comment]
@msfussell The actions be canceled when installing docker on macos. Can u help me re-run it?

**Comment by amulyavarote** [comment]
@noahziheng PR with build fix is merged. Can you please rebase the changes and push?

**Comment by noahziheng** [comment]
@amulyavarote done~

**Review by amulyavarote — APPROVED** [review]

**Pull request merged**